Framework/PythonInterface/plugins/algorithms/FindGoniometerFromUB.py

```diff
@@ -0,0 +1,293 @@
+            doc="Tolerance of chi angle to test for consistency.")
+        self.declareProperty(
+            name="PhiTol",
+            defaultValue=15.0,
+            validator=FloatBoundedValidator(
+                lower=0.0, upper=45.0),
+            direction=Direction.Input,
+            doc="Tolerance of phi angle to test for consistency (due to uncertainty on the gonio axis at omega=0"
+                "this tol might need to be quite large).")
+        self.declareProperty(
+            name="PhiHand",
+            defaultValue=1,
+            validator=IntListValidator([-1, 1]),
+            direction=Direction.Input,
+            doc="Handedness for the phi angles (rotation around goniometer axis) in the"
+                " log file (1 for ccw/RH, -1 for cw/LH)")
+        self.declareProperty(
+            name="DOmega",
+            defaultValue=0.0,
+            direction=Direction.Input,
+            validator=FloatBoundedValidator(
+                lower=0.0, upper=180.0),
+            doc="omega rotation between component of goniometer axis in XY-plane from +ve Y-axis"
+                "(perpendicular to ki)")
+        self.declareProperty(
+            name="OmegaHand",
+            defaultValue=1,
+            validator=IntListValidator([-1, 1]),
+            direction=Direction.Input,
+            doc="Handedness for the omega angles (rotation around +ve Z) in the log file (1 for ccw/RH, -1 for cw/LH)")
+        # output
+        self.declareProperty(
+            ITableWorkspaceProperty(
+                name="OutputTable",
+                defaultValue="",
+                validator=StringMandatoryValidator(),
+                direction=Direction.Output),
+            doc="Output table to display goniometer axis and angles for runs with UBs that can share common indexing")
+
+    def validateInputs(self):
+        issues = dict()
+        # UBpath must contain more than one run
+        ubFiles = self.getProperty('UBfiles').value
+        if len(ubFiles) < 2:
+            issues['UBfiles'] = "At least two UB files are required"
+        return issues
+
+    def PyExec(self):
+
+        # setup progress bar
+        prog_reporter = Progress(self, start=0.0, end=1.0, nreports=3)
+        # Get input
+        ubFiles = self.getProperty('UBfiles').value
+        chiRef = self.getProperty('Chi').value
+        chiTol = self.getProperty('ChiTol').value
+        phiTol = self.getProperty('PhiTol').value
+        phiHand = self.getProperty('PhiHand').value
+        dOmega = self.getProperty('DOmega').value
+        omegaHand = self.getProperty('OmegaHand').value
+        # rotation matrices work in right-handed coordinate system so the handedness
+
+        # self.log().information(msg)
+        # MultipleFileProperty -> UB matrix (first one for reference)
+
+        # update progress
+        prog_reporter.report(1, "Load Logs and UBs")
+
+        matUB = []  # container to hold UB matrix arrays
```

Comment:
Perhaps consider creating a separate function for this `for` loop so you can have:
```python
matUB, omega, phiRef = self.load_ubfiles()
```